src/ledger.js

```diff
@@ -0,0 +1,131 @@
+// LedgerSystem class
+const BigNumber = require('bignumber.js');
+const Data = require("./database");
+const FIFOQueue = require("./queue");
+const sleep = require('./utils').sleep;
+const groupBy = require('./utils').groupBy;
+const OperationTypes = {
+    TRANSFER: "TRANSFER"
+}
+const OperationStatus = {
+    INIT: "INIT",
+    PROCESSING: "PROCESSING",
+    APPLIED: "APPLIED",
+    REJECTED: "REJECTED"
+}
+const Entities = {
+    OPERATIONS: "operations",
+    ACCOUNTS: "accounts",
+    ENTRIES: "entries"
+}
+class LedgerSystem {
+    // singleton class
+    constructor (databaseConfig) {
+        this.db = new Data[databaseConfig.class](databaseConfig);
+        this.postingQueue = new FIFOQueue(this.db.getAll(Entities.OPERATIONS).filter(operation => [OperationStatus.INIT, OperationStatus.PROCESSING].includes(operation.status)).sort((a, b) => a.id - b.id).map((operation) => {
```

Comment:
Please break this down and make it more readable

Reply:
Intermediate variables are our friends, they help make such long chains of logic more readable.